### PR TITLE
Fix print function and strings issue

### PR DESCRIPTION
**Pull request by maliasadi**
Fix https://github.com/PennyLaneAI/catalyst/issues/404

[sc-52664]

### CODE REVIEW

**Review by erick-xanadu — APPROVED** [review]
Love it!

**Comment by erick-xanadu** [comment]
@maliasadi just make sure that you test it similarly to the issue. Does python treat "\0" as a string or as a character?

File: doc/changelog.md [hunk 1]
```diff
@@ -152,6 +152,10 @@
   if a gradient pass cannot successfully eliminate all quantum operations for such functions.
   [(#397)](https://github.com/PennyLaneAI/catalyst/issues/397)
 
+* Resolve a bug with printing multiple strings from multiple functions with `debug.print` by introducing
+  a NULL terminator to traced strings.
+  [(#418)](https://github.com/PennyLaneAI/catalyst/pull/418)
```
[the remark this answers is not in the document]
Comment (maliasadi) [reply doc/changelog.md:157]: @josh146 Added [590ae50](https://github.com/PennyLaneAI/catalyst/pull/418/commits/590ae500bd9365ce214b598fe36ce0f463af7608)

**Comment by maliasadi** [comment]
> Does python treat "\0" as a string or as a character?

In Python, it represents as a string of length 1 containing a single null char:

``` python
>>> null_str = "\0"
>>> len(null_str)
1
>>> null_str[0]
'\x00'
```

**Review by dime10 — APPROVED** [review]

**Pull request merged**